packages/lodestar/test/e2e/node.test.ts

```diff
@@ -8,7 +8,7 @@ describe("beacon node", function () {
 
   it("should justify block", async function () {
     this.timeout(120000);
-    const bn = await getDevBeaconNode({SECONDS_PER_SLOT: 2, SLOTS_PER_EPOCH: 8});
+    const bn = await getDevBeaconNode({SECONDS_PER_SLOT: 2, SLOTS_PER_EPOCH: 8}, {sync: {minPeers: 0}});
```

Comment:
Should work without minPeers: 0 since you don't need to gossip if you don't have peers :slightly_smiling_face:

Reply:
removing this causes bugs, stemming from our sync attestation collector trying to interact with a stopped gossipsub.

Will fix in this PR, since it seems related.

Reply:
I think having gossip on (eg; being fully synced on a chain) is a key part of a validator running. If not, the validator's attestations will not be broadcast properly to the network.
In this case, I don't know where the boundary should be, but it is not good to have gossip off while the chain is being driven by validators.

Recommendation is to leave these changes as-is in this PR, make an issue about syncing status and validator input.
Eg: "validator processing should be dependent on syncing status." or something to that effect.
Perhaps there are two separable sub-issues.
One on the vc side: "validator client should initially check the bn sync status"
One on the bn side: "validator api should handle data received during unsynced chain"

Reply:
Actually, we should have middleware to return 403 or something from validator related api if node is syncing.

Can we make attestation collector to silently ignore errors if gossip sub is off. I mean there is no point in it if we don't have enough peers anyways?

Reply:
It likely shouldn't be a middleware, because we will want the same behavior in the JS api.

I think we can do "the log.error then return" in `sync.collectAttestations`. We should also do the sync check on the API level since that should throw an error, instead of just return.

Reply:
Just changed this to throw an Error in `sync.collectAttestations` if the chain isn't in regular sync or synced mode. I think it will be better to throw an error here than silently return.
Sync is an internal module, so we know that anything that attempts to run `sync.collectAttestations` without the chain being in regular sync mode is an error.